lib/ecto/changeset.ex

```diff
@@ -1,14 +1,17 @@
 defmodule Ecto.Changeset do
   @moduledoc ~S"""
-  Changesets allow filtering, casting, validation and
-  definition of constraints when manipulating structs.
-
-  There is an example of working with changesets in the introductory
-  documentation in the `Ecto` module. The functions `cast/4` and
-  `change/2` are the usual entry points for creating changesets.
-  The first one is used to cast and validate external parameters,
-  such as parameters sent through a form, API, command line, etc.
-  The second one is used to change data directly from your application.
+  Changesets allow filtering, type conversion, validation and
```

Comment:
```suggestion
  Changesets allow filtering, type casting, validation and
```

Let's please either call it "casting" or "type casting", as that's the term we consistently use through the docs. We already detail it below now, which should be enough.

Reply:
If you don't want this I obviously can't make you, but I will just say again, in different words: this is trying to be technically precise in the part of the documentation that will be most seen by people who need a 5th grade English description of the purpose of this file, and be glossed over by the people who have graduated to using it purely as a reference.

Also it is referred to as type conversion in the External section and in several other places in the docs, and a bunch of places in the issues database. I didn't make it up, or introduce it, I borrowed from what is already there.

Reply:
Correct. It is ok if we expand and use alternative names later on. But this is literally the one sentence summary. We should be precise and use appropriate naming. The breakdown comes later.

Reply:
Another way to put it down: the first sentence is what will briefly recap what you read in the past, when you are browsing through the docs or when you need to remember exactly what this module does. It is unreasonable for the audience you describe to expect them to read the first sentence and be like "oh, I know exactly what it does". They have to read the rest of the document.

Reply:
> Another way to put it down: the first sentence is what will briefly recap what you read in the past, when you are browsing through the docs or when you need to remember what this module does, and casting is a very 

Introductions are meant to be vague and refined as you go. 'Recapping' a term that hasn't been defined yet is not recapping.

"What is a changeset?" now requires me to learn what casting is. Which is probably okay if "What is a changeset?" isn't already the second or third question in a chain of terminology yak shaving.

But that's exactly where I was when I was reading this section, and exactly the sorts of conversations I've had with other people making similar edits to internal company docs. Many, many times. 

I'll change it back, but I'm doing it under duress.  This revert undermines most of the changes made further down.